src/mongo/db/encryption/key_operations.h

```diff
@@ -0,0 +1,317 @@
+/*======
+This file is part of Percona Server for MongoDB.
+
+Copyright (C) 2022-present Percona and/or its affiliates. All rights reserved.
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the Server Side Public License, version 1,
+    as published by MongoDB, Inc.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    Server Side Public License for more details.
+
+    You should have received a copy of the Server Side Public License
+    along with this program. If not, see
+    <http://www.mongodb.com/licensing/server-side-public-license>.
+
+    As a special exception, the copyright holders give permission to link the
+    code of portions of this program with the OpenSSL library under certain
+    conditions as described in each individual source file and distribute
+    linked combinations including the program with the OpenSSL library. You
+    must comply with the Server Side Public License in all respects for
+    all of the code used other than as permitted herein. If you modify file(s)
+    with this exception, you may extend this exception to your version of the
+    file(s), but you are not obligated to do so. If you do not wish to do so,
+    delete this exception statement from your version. If you delete this
+    exception statement from all source files in the program, then also delete
+    it in the license file.
+======= */
+
+#pragma once
+
+#include <cstdint>
+#include <memory>
+#include <optional>
+#include <string>
+#include <string_view>
+
+#include "mongo/db/encryption/key_id.h"
+
+namespace mongo {
+class EncryptionGlobalParams;
+
+namespace encryption {
+class Key;
+
+/// @breif The operation of reading an encryption key from a key management
+/// facility.
+///
+/// Examples of a key management facility include an encrytion key file,
+/// a Vault server, and a KMIP server. Individual implementations encapsulate
+/// how a key is read from a particular facility.
+struct ReadKey {
```

Comment:
Let's call this 'class' instead of 'struct'

Reply:
fixed